lambdaclass/mirra_backend
# [GH-360] [CONFIG] Kaline Tree config

## Motivation

Adds Kaline Tree level config.
Closes https://github.com/lambdaclass/afk_gacha_game/issues/360

When I ran into some test issues, I found this smelly approach we took for test-iterating values on our tests where we were running `assert Enum.all?` to the enumerable we wanted to check. This left us with not-nice, not-descriptive assertion errors. I've refactored them into `Enum.each` calls with assertions at the end. Now test errors look nicer! This brought up a lot of unused variable/alias/import warnings which I also handled.

## Summary of changes

- Make KT level up costs dynamic.
- Add and implement experience afk reward for KT. 
- Add KT config.
- Refactor champions tests.


## How to test it?

Game should run and work just fine through the client! Also, check that you can now get XP by claiming afk rewards.
